c7n/resources/rds.py

```diff
@@ -1568,3 +1568,103 @@ def process(self, resources, event=None):
+                  - DeletionProtection: true
+                  - PubliclyAccessible: true
+                actions:
+                  - type: modify-db
+                    update:
+                      - property: 'DeletionProtection'
+                        value: false
+                      - property: 'PubliclyAccessible'
+                        value: false
+                    immediate: true
+    """
+
+    schema = type_schema(
+        'modify-db',
+        immediate={"type": 'boolean'},
+        update={
+            'type': 'array',
+            'items': {
+                'type': 'object',
+                'properties': {
+                    'property': {'type': 'string', 'enum': [
+                        'AllocatedStorage',
+                        'DBInstanceClass',
+                        'DBSubnetGroupName',
+                        'DBSecurityGroups',
+                        'VpcSecurityGroupIds',
+                        'MasterUserPassword',
+                        'DBParameterGroupName',
+                        'BackupRetentionPeriod',
+                        'PreferredBackupWindow',
+                        'PreferredMaintenanceWindow',
+                        'MultiAZ',
+                        'EngineVersion',
+                        'AllowMajorVersionUpgrade',
+                        'AutoMinorVersionUpgrade',
+                        'LicenseModel',
+                        'Iops',
+                        'OptionGroupName',
+                        'NewDBInstanceIdentifier',
+                        'StorageType',
+                        'TdeCredentialArn',
+                        'TdeCredentialPassword',
+                        'CACertificateIdentifier',
+                        'Domain',
+                        'CopyTagsToSnapshot',
+                        'MonitoringInterval',
+                        'DBPortNumber',
+                        'PubliclyAccessible',
+                        'DomainIAMRoleName',
+                        'PromotionTier',
+                        'EnableIAMDatabaseAuthentication',
+                        'EnablePerformanceInsights',
+                        'PerformanceInsightsKMSKeyId',
+                        'PerformanceInsightsRetentionPeriod',
+                        'CloudwatchLogsExportConfiguration',
+                        'UseDefaultProcessorFeatures',
+                        'DeletionProtection']},
+                    'value': {}
+                },
+            },
+        },
+        required=('update',))
+
+    permissions = ('rds:ModifyDBInstance',)
+
+    def process(self, resources):
+        c = local_session(self.manager.session_factory).client('rds')
+
+        param = {}
+        param['ApplyImmediately'] = self.data.get('immediate', False)
+
+        for x in self.data.get('update'):
+            param[x['property']] = x['value']
+
+        for r in resources:
+            param['DBInstanceIdentifier'] = r['DBInstanceIdentifier']
```

Comment:
the other thing that would be nice here is not modifying resources that have the same value thats being set.

Reply:
almost, technically the minimal modification set here, is making a copy of param dict per resource, and removing all keys that have duplicated values. at the moment its skipping only when all values are the same.

Reply:
That makes sense, I was trying to save space by not copying dicts and assumed that if I pass in a modify call with parameters that are the same as the current rds instance, it would simply not update those parameters by having some sort of check. I'm assuming thats not true and will still take time/resources to modify the rds instance and redundantly update it. Is this true?

Reply:
either way, updated it